python/lsst/ip/isr/isrFunctions.py

```diff
@@ -527,13 +527,27 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
 
         if overscanIsInt:
             del overscanImageI
-    elif fitType in ('POLY', 'CHEB', 'LEG', 'NATURAL_SPLINE', 'CUBIC_SPLINE', 'AKIMA_SPLINE'):
-        if hasattr(overscanImage, "getImage"):
-            biasArray = overscanImage.getImage().getArray()
-            biasArray = numpy.ma.masked_where(overscanImage.getMask().getArray() & statControl.getAndMask(),
+    elif fitType in ('MEDIAN_PER_ROW', 'POLY', 'CHEB', 'LEG', 'NATURAL_SPLINE', 'CUBIC_SPLINE',
+                     'AKIMA_SPLINE'):
+        if fitType in ('MEDIAN_PER_ROW',):
+            # As above, we need an image with integers to calculate the median per row
+            if overscanIsInt:
+                if hasattr(overscanImage, "image"):
+                    imageI = overscanImage.image.convertI()
+                    overscanImageI = afwImage.MaskedImageI(imageI, overscanImage.mask, overscanImage.variance)
+                else:
+                    overscanImageI = overscanImage.convertI()
+            else:
+                overscanImageI = overscanImage
+        else:
+            overscanImageI = overscanImage
```

Comment:
This giant cascade of `elif if if if / else else else` is horrendous! It's not your fault at all, this code was a mess when you arrived, but I think it desperately needs refactoring. Can you file a ticket to refactor this whole function please? (You don't have to do the ticket, of course)

Reply:
Done: https://jira.lsstcorp.org/browse/DM-23396

Reply:
(and yes, I agree that it is not elegant)

Reply:
Great, thanks!